ElrondNetwork/elrond-go
# On API, fix GetAccountWithBlockInfo()

## Description of the reasoning behind the pull request (what feature was missing / how the problem was manifesting itself / what was the motive behind the refactoring)
- When running `rosetta-ci` (continuous validation, private repository), we've detected a number of balance reconciliation issues that indicated a data inconsistency on high load (observer syncing, rosetta-cli making many GET account requests).
- Upon code inspection, we've detected that in function `GetAccountWithBlockInfo()`, more goroutines can concur to recreate the trie, but the returned `blockInfo` did not reflect those trie re-creations.
  
## Proposed Changes
- In `accountsDBApi.go`, function `GetAccountWithBlockInfo()`: return the `blockInfo` tied to the current trie within the `innerAccountsAdapter` (member field). This way, the returned `blockInfo` is correctly linked to the returned account state.

## Testing procedure
- Run rosetta-cli's `check:data` validation on Devnet (high load, with number of altered accounts per block larger than concurrency of active reconciliation)
